var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -212,6 +213,9 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("kokkos-nvcc-wrapper@master", when="@master+wrapper")
     conflicts("+wrapper", when="~cuda")
 
+    conflicts("+rocthrust", when="@4.3: ~rocm")
+    conflicts("~rocthrust", when="@4.3: +rocm")
```

Comment:
```suggestion
```

After more thought, always enabling `rocthrust` when enabling `room` makes more sense for the users. And it is more straightforward.